src/dapp/extensionProvider.ts

```diff
@@ -0,0 +1,214 @@
+    }
+    this.extensionPopupWindow = window.open(
+      this.extensionURL,
+      this.popupName,
+      this.popupOptions
+    );
+  }
+
+  private startBgrMsgChannel(
+    operation: string,
+    connectData: any
+  ): Promise<any> {
+    return new Promise((resolve, reject) => {
+      window.postMessage(
+        {
+          target: "erdw-inpage",
+          type: operation,
+          data: connectData,
+        },
+        window.origin
+      );
+
+      const eventHandler = (event: any) => {
+        if (
+          event.isTrusted &&
+          event.data.type &&
+          event.data.target === "erdw-contentScript"
+        ) {
+          switch (event.data.type) {
+            case "logoutResponse":
+              window.removeEventListener("message", eventHandler);
+              resolve(true);
+              break;
+          }
+        }
+      };
+      setTimeout(() => {
+        reject(
+          "Extension logout response timeout. No response from extension."
+        );
+      }, 3000);
+      window.addEventListener("message", eventHandler, false);
+    });
+  }
+
+  private startExtMsgChannel(operation: string, connectData: any): any {
+    return new Promise((resolve, reject) => {
+      var isResolved = false;
+      const eventHandler = (event: any) => {
+        if (
+          event.isTrusted &&
+          event.data.type &&
+          event.data.target === "erdw-extension"
+        ) {
+          switch (event.data.type) {
+            case "popupReady":
+              event.ports[0].postMessage({
+                target: "erdw-inpage",
+                type: operation,
+                data: connectData,
+              });
+              break;
+            case "connectResult":
+              this.extensionPopupWindow?.close();
+              this.account = event.data.data;
+              window.removeEventListener("message", eventHandler);
+              isResolved = true;
+              resolve(event.data.data);
+              break;
+
+            default:
+              this.handleExtResponseErr(event);
+              this.extensionPopupWindow?.close();
+              window.removeEventListener("message", eventHandler);
+              isResolved = true;
+              resolve(event.data.data);
+              break;
+          }
+        }
+      };
+      var windowCloseInterval = setInterval(() => {
+        if (this.extensionPopupWindow?.closed) {
+          window.removeEventListener("message", eventHandler);
+          clearInterval(windowCloseInterval);
+          if (!isResolved)
+            reject("Extension window was closed without response.");
+        }
+      }, 500);
+
+      window.addEventListener("message", eventHandler, false);
+    });
+  }
+
+  private handleExtResponseErr(event: any) {
+    if (!event.data || !event.data.data) {
+      if (
+        event.data.type === "transactionComplete" &&
+        event.data.data.length === 0
+      )
```

Comment:
Can we use brackets on all `if` conditions? It will be aligned with the rest of the code, also my linter does not like the missing `{}`